### PR TITLE
Let integration-tests know about latest WP versions

### PR DESCRIPTION
**Pull request by szepeviktor**
@johnbillion So you can say
```yaml
wp:
  - latest
  - latest-1
  - latest-2
```
and **not** update repos using this workflow every time.

### CODE REVIEW

File: .github/workflows/reusable-integration-tests.yml [hunk 1]
```diff
@@ -69,6 +69,10 @@ jobs:
       run: |
         if [ "${{ inputs.wp }}" == "latest" ]; then
           composer install --dev --prefer-dist
```
Comment (szepeviktor) [inline .github/workflows/reusable-integration-tests.yml:71]: Explicit is better than implicit.
```
composer require --dev --update-with-dependencies --prefer-dist roots/wordpress-full="~6.5.0"
```

**Comment by johnbillion** [comment]
Over the years I have gone back and forth between using explicit version numbers and some sort of "latest" identifier.

The problem with using a system like `latest`, `latest-1`, `latest-2` is that the consuming plugin may not yet be tested with the latest version, for example on the day that WordPress 6.6 is released the workflows may start failing because I haven't yet made the plugin compatible with that version. By using explicit version numbers in the workflows for each plugin I can control exactly when I update them and can control the tested versions in the PR for the update.

Thanks as always Viktor but I'll close this off. I'm happy updating the explicit version numbers in the workflows for each plugin.

**Comment by szepeviktor** [comment]
> the workflows may start failing because I haven't yet made the plugin compatible

I thought `nightly` is for preparing for the next version.

**Comment by johnbillion** [comment]
Yeah it is, but that doesn't mean that any given plugin is ready and fully tested on the day a new major version of WordPress is released.

Honestly I'm happy managing this within each plugin. It's not a huge amount of work and it's usually accompanied by other changes such as bumping the minimum supported version, checking that actions are all up to date, etc etc.

**Pull request closed without merging**